docs/Using-Inbox-Service.md

```diff
@@ -0,0 +1,151 @@
+# Using Inbox Service
+
+<!-- begin remove -->
+- [Introduction](#introduction)
+- [Creating an Instance](#creating-an-instance)
+- [Inbox Service Usage](#inbox-service-usage)
+  - [Get Number of Unread Messages](#get-number-of-unread-messages)
+  - [Get List of Messages](#get-list-of-messages)
+  - [Get Message Detail](#get-message-detail)
+  - [Set Message as Read](#set-message-as-read)
+- [Error handling](#error-handling)
+<!-- end -->
+
+## Introduction
+
+Inbox Service is responsible for managing messages in the Inbox. The inbox is a simple one way delivery system that allows you to deliver messages to the user.
+
+<!-- begin box warning -->
+Note: Before using Inbox Service, you need to have a `PowerAuthSDK` object available and initialized with a valid activation. Without a valid PowerAuth activation, service will return an error.
+<!-- end -->
+
+## Creating an Instance
+
+### On Top of the `PowerAuthSDK` instance
+```swift
+import WultraMobileTokenSDK
+import WultraPowerAuthNetworking
+
+let networkingConfig = WPNConfig(
+    baseUrl: URL(string: "https://myservice.com/mtoken/push/api/")!,
+    sslValidation: .default
+)
+// powerAuth is instance of PowerAuthSDK
+let inboxService = powerAuth.createWMTInbox(networkingConfig: networkingConfig)
+```
+
+### On Top of the `WPNNetworkingService` instance
+```swift
+import WultraMobileTokenSDK
+
+// networkingService is instance of WPNNetworkingService
+let inboxService = networkingService.createWMTInbox()
+```
+
+## Inbox Service Usage
+
+### Get Number of Unread Messages
+
+To get the number of unread messages, use the following code:
+
+```swift
+inboxService.getUnreadCount { result in
+    switch result {
+    case .success(let count):
+        if count.countUnread > 0 {
+            print("There are \(count.countUnread) new message(s) in your inbox")
+        } else {
+            print("Your inbox is empty")
+        }
+    case .failure(let error):
+        print("Error \(error)")
+    }    
+}
+```
+
+### Get List of Messages
+
+To get the list of all messages, call:
+
+```swift
+inboxService.getAllMessages { result in 
```

Comment:
This should be discouraged a bit more - I think that we should make the "paged approach" the default one, and this can be in "Other topics"...